src/View/Helper/UserHelper.php

```diff
@@ -167,11 +167,26 @@ public function addReCaptcha()
             );
         }
         $this->addReCaptchaScript();
-        try {
-            $this->Form->unlockField('g-recaptcha-response');
-        } catch (\Exception $e) {
+        $version = Configure::read('Users.reCaptcha.version',2);
+        $method = "addReCaptchaV$version";
+        if (method_exists($this, $method)) {
+            try {
+                $this->Form->unlockField('g-recaptcha-response');
+            } catch (\Exception $e) {
+            }
+            return $this->{$method}();
         }
+        throw new \InvalidArgumentException(__d('cake_d_c/users', 'reCaptcha version is wrong. Please configure Users.reCaptcha.version as 2 or 3'));
+
+    }
 
+    /**
+     * Add required element for reCaptcha v2
+     *
+     * @return string
+     */
+    private function addReCaptchaV2()
```

Comment:
let's add a deprecation warning in case of V2 and switch default to V3 in the next major version?